src/server/worker/master.go

```diff
@@ -531,366 +526,241 @@ func plusDuration(x *types.Duration, y *types.Duration) (*types.Duration, error)
-							if skipped {
-								// write a list of input files
-								if err := statsTree.PutFile(fmt.Sprintf("%v/skipped", datumID), nil, 0); err != nil {
-									logger.Logf("failed to write skipped file, this is non-fatal but will result in some missing stats")
-									return nil
+							high := high
+							eg.Go(func() error {
+								for i := low; i < high; i++ {
+									i := i
+									limiter.Acquire()
+									eg.Go(func() error {
+										defer limiter.Release()
+										files := df.Datum(int(i))
+										return a.collectDatum(ctx, int(i), files, logger, tree, statsTree, &treeMu, chunkState.State == ChunkState_FAILED)
+									})
 								}
-							}
-							// Add a file to statsTree indicating the index of this
-							// datum in the datum factory.
-							if err := statsTree.PutFile(fmt.Sprintf("%v/index", datumID), []*pfs.Object{indexObject}, length); err != nil {
-								logger.Logf("failed to write index file, this is non-fatal but will result in some missing stats")
 								return nil
-							}
-							return nil
-						})
-					}
-					if err := eg.Wait(); err != nil {
-						return err
-					}
-					treeMu.Lock()
-					defer treeMu.Unlock()
-					if statsSubtree != nil {
-						if err := statsTree.Merge(statsSubtree); err != nil {
-							logger.Logf("failed to merge into stats tree: %v", err)
+							})
+							break EventLoop
 						}
-					}
-					if stats != nil {
-						processStats = append(processStats, stats)
-					}
-					return tree.Merge(subTree)
-				}, b, func(err error, d time.Duration) error {
-					select {
 					case <-ctx.Done():
-						return err
-					default:
-					}
-					if userCodeFailures > maximumRetriesPerDatum {
-						logger.Logf("job %s failed to process datum %+v %d times failing", jobID, files, userCodeFailures)
-						failed = true
-						return err
-					}
-					logger.Logf("job %s failed to process datum %+v with: %+v, retrying in: %+v", jobID, files, err, d)
-					return nil
-				}); err == nil {
-					if skipped {
-						go updateProgress(0, 1, stats)
-					} else {
-						go updateProgress(1, 0, stats)
+						return context.Canceled
 					}
 				}
-			}()
+				return nil
+			}(); err != nil {
+				return err
+			}
+		}
+		if err := eg.Wait(); err != nil {
+			return err
 		}
-		limiter.Wait()
 
 		var statsCommit *pfs.Commit
 		if jobInfo.EnableStats {
-			if err := func() error {
-				aggregateProcessStats, err := a.aggregateProcessStats(processStats)
-				if err != nil {
-					return err
-				}
-				marshalled, err := (&jsonpb.Marshaler{}).MarshalToString(aggregateProcessStats)
-				if err != nil {
-					return err
-				}
-				aggregateObject, _, err := a.pachClient.WithCtx(ctx).PutObject(strings.NewReader(marshalled))
-				if err != nil {
-					return err
-				}
-				return statsTree.PutFile("/stats", []*pfs.Object{aggregateObject}, int64(len(marshalled)))
-			}(); err != nil {
-				logger.Logf("error aggregating stats")
-			}
+			// if err := func() error {
```

Comment:
Should we delete all of this commented-out code?

Reply:
Done.